code/lib/preview-api/src/modules/store/csf/beforeAll.ts

```diff
@@ -0,0 +1,21 @@
+type Awaitable<T> = T | Promise<T>;
+
+type CleanupFn = () => Awaitable<void>;
+
+export type BeforeAllHook = () => Awaitable<void | CleanupFn>;
+
+// Execute all the hooks in sequence, and return a function that will execute cleanups in reverse order
+export const composeBeforeHooks = (hooks: BeforeAllHook[]): BeforeAllHook => {
+  return async () => {
+    const cleanups: CleanupFn[] = [];
+    for (const hook of hooks) {
+      const cleanup = await hook();
+      if (cleanup) cleanups.push(cleanup);
+    }
+    return async () => {
+      for (const cleanup of cleanups.reverse()) {
```

Comment:
Maybe to be sure, [...cleanups.reverse()] as reverse is mutable.
Although technically, this should not be called twice.

Reply:
Sure. I've replaced `push` with `unshift` so we avoid reversing altogether.